camel-core/src/main/java/org/apache/camel/ComponentVerifier.java

```diff
@@ -21,118 +21,287 @@
             }
-
             throw new IllegalArgumentException("Unknown scope <" + scope + ">");
         }
     }
 
+    // =============================================================================================
+
     /**
-     * Represent an error
+     * This interface represents a detailed error in case when the verification fails.
      */
-    interface Error extends Serializable {
+    interface VerificationError extends Serializable {
+
         /**
-         * @return the error code
+         * The overall error code, which can be either a {@link StandardCode} or a custom code. It is
+         * recommended to stick to the predefined standard codes
+         *
+         * @return the general error code.
          */
-        String getCode();
+        Code getCode();
 
         /**
+         * A human readable description of the error in plain english
+         *
          * @return the error description (if available)
          */
         String getDescription();
 
         /**
-         * @return the parameters in error
+         * A set of input parameter names which fails the verification. These are keys to the parameter provided
+         * to {@link #verify(Scope, Map)}.
+         *
+         * @return the parameter names which are malformed and caused the failure of the validation
          */
-        Set<String> getParameters();
+        Set<String> getParameterKeys();
 
         /**
-         * @return a number of key/value pair with additional information related to the validation.
+         * Details about the failed verification. The keys can be either predefined values
+         * ({@link ExceptionAttribute}, {@link HttpAttribute}, {@link GroupAttribute}) or it can be free-form
+         * custom keys specific to a component. The standard attributes are defined as enums in all uppercase (with
+         * underscore as separator), custom attributes are supposed to be in all lower case (also with underscores
+         * as separators)
+         *
+         * @return a number of key/value pair with additional information related to the verification.
          */
-        Map<String, Object> getAttributes();
-    }
+        Map<Attribute, Object> getDetails();
 
-    /**
-     * Represent a validation Result.
-     */
-    interface Result extends Serializable {
-        enum Status {
-            OK,
-            ERROR,
-            UNSUPPORTED
+        /**
+         * Interface defining an error code. This is implemented by the {@link StandardCode} but also
+         * own code can be generated by implementing this interface. This is best done via {@link ResultErrorBuilder#asCode(String)}
+         * If possible, the standard codes should be reused
+         */
+        interface Code {
+            /**
+             * Name of the code. All uppercase for standard codes, all lower case for custom code. Separator is underscore.
+             * @return code name
+             */
+            String name();
```

Comment:
Wondering if this need to have an ```equals``` as if you do not have access to the Attribute instance you may need to construct a new Attribute via ```asAttribute(String)``` which then is not found in the map.